### ISSUE TITLE
RETURNING with composite primary keys fails

### ISSUE THREAD

**Issue by Overbryd** [jruby/activerecord-jdbc-adapter#773]
I am using `activerecord-jdbcpostgresql-adapter` in one of my projects. And the problem my tests stumbled over (not a problem using CRuby with `pg`) is the following:

```
activerecord (4.2.7.1)
activerecord-jdbc-adapter (1.3.22)
activerecord-jdbcpostgresql-adapter (1.3.22)
jruby 9.1.7.0 (2.3.1) 2017-01-11 68056ae Java HotSpot(TM) 64-Bit Server VM 25.45-b02 on 1.8.0_45-b14 +jit [darwin-x86_64]
```

I insert on a join table, its primary key is a composite of two columns `collection_id` and `product_id`.

The insert it generates looks like this:

```sql
INSERT INTO "collections_products" ("product_id", "collection_id")
VALUES (1, 1)
RETURNING "collection_id,product_id"
```

_I added some newlines for readability._

If you look at the `RETURNING` clause, you can spot the error, as it is not properly converting these values to columns. And PostgreSQL throws me an `ERROR: column "collection_id,product_id" does not exist`.

I expect it to look something like this, with parantheses `(` around the two columns `)`:

```sql
INSERT INTO "collections_products" ("product_id", "collection_id")
VALUES (1, 1)
RETURNING ("collection_id", "product_id")
```

Currently I am digging in the source to find the problem, but so far no luck.

**If you know a quicker answer, please help me out.**

**Comment by Overbryd** [comment]
I found the culprit.

```ruby
# lib/arjdbc/postgresql/adapter.rb:566

    def sql_for_insert(sql, pk, id_value, sequence_name, binds)
      unless pk
        # Extract the table from the insert sql. Yuck.
        table_ref = extract_table_ref_from_insert_sql(sql)
        pk = primary_key(table_ref) if table_ref
      end

      if pk && use_insert_returning?
        sql = "#{sql} RETURNING #{quote_column_name(pk)}"
      end

      [ sql, binds ]
    end
```

It is using `quote_column_name(pk)` but should be using something more sophisticated that takes composite pk into account. `pk` here is an array that looks like this: `["collection_id", "product_id"]`.

**Comment by kares** [comment]
we do not have the error message but #756 should be a duplicate

**Comment by kares** [comment]
closed with #774 ... thanks
